File: 0717. 1-bit and 2-bit Characters/Solution.cpp
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        bool expectNext = false;
        
        for(int i=0;i<bits.size();i++)
        {
            expectNext = false;
            if(bits[i] == 1)
            {
                expectNext = true;
                i++;
            }
        }
        
        return !expectNext;
    }
};

File: 0717. 1-bit and 2-bit Characters/Solution.py
class Solution {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        // Remove trailing zero
        // bits.pop_back();
        int i;
        
        for(i=0;i<bits.size()-1;i++)
        {
            i += bits[i];
        }
        
        return i == bits.size()-1;
        
    }
};
